tools/snp.sh

```diff
@@ -786,13 +815,21 @@ setup_and_launch_guest() {
   #add_qemu_cmdline_opts "-machine pc-q35-7.1"
 
   # snp object and kernel-hashes on
-  add_qemu_cmdline_opts "-object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,kernel-hashes=on"
+  if "$SVSM"; then
+     add_qemu_cmdline_opts "-object memory-backend-memfd,size=2G,id=mem0,share=true,prealloc=false,reserve=off"
+     add_qemu_cmdline_opts "-object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,init-flags=5,igvm-file="$IGVM_FILE""
+  else
+     add_qemu_cmdline_opts "-object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1,kernel-hashes=on"  
+  fi
 
   # ovmf, initrd, kernel and append options
-  add_qemu_cmdline_opts "-bios ${OVMF_BIN}"
-  add_qemu_cmdline_opts "-initrd ${INITRD_BIN}"
-  add_qemu_cmdline_opts "-kernel ${KERNEL_BIN}"
-  add_qemu_cmdline_opts "-append \"${GUEST_KERNEL_APPEND}\""
+  if "$SVSM"; then
+    add_qemu_cmdline_opts "-initrd ${INITRD_BIN}"
+    add_qemu_cmdline_opts "-kernel ${KERNEL_BIN}"
+    add_qemu_cmdline_opts "-append \"${GUEST_KERNEL_APPEND}\""
+  else
+    add_qemu_cmdline_opts "-bios ${OVMF_BIN}"
+  fi
```

Comment:
This seems backward to me?

Reply:
Thanks. The changes have been corrected now.